spec/models/storage_provider_spec.rb

```diff
@@ -32,33 +33,61 @@
 
     describe '#store_aws' do
       before do
-        allow(storage_provider).to receive(:s3_uploader).and_return(s3_uploader)
+        allow(storage_provider).to receive(:aws_s3_uploader).and_return(aws_s3_uploader)
       end
 
       let(:expected_metadata) { { 'a' => 'b' } }
       let(:expected_tags) { { 'c' => 'd' } }
 
       it 'performs as expected' do
-        expect(s3_uploader).to receive(:upload_file).with(
+        expect(aws_s3_uploader).to receive(:upload_file).with(
           pending_transfer.source_object.path,
-          s3_object_key,
+          object_key,
           :whole_file,
           overwrite: false,
           metadata: expected_metadata,
           tags: expected_tags,
           precalculated_aws_crc32c: 'm6nHZg=='
         )
-        storage_provider.store_aws(pending_transfer, s3_object_key, metadata: expected_metadata, tags: expected_tags)
+        storage_provider.store_aws(pending_transfer, object_key, metadata: expected_metadata, tags: expected_tags)
```

Comment:
these specs should probably exercise #perform_transfer (what the job calls) and not the provider-specific method (or at least, in addition to that)

Reply:
Good call, thank you.